src/main/java/org/spongepowered/common/command/SpongeCommandManager.java

```diff
@@ -299,6 +299,9 @@ public boolean containsMapping(CommandMapping mapping) {
 
     @Override
     public CommandResult process(CommandSource source, String commandLine) {
+        if (!Sponge.getServer().isMainThread()) {
```

Comment:
Wouldn't it be better to use SpongeImplHooks#isMainThread ?

Reply:
I don't really think it matters either way, but it doesn't hurt to be defensive.

Reply:
It does matter in the sense that the `SpongeImplHooks#isMainThread` has a stupid fast optimization check (much like the phase tracker)

Reply:
I'm actually curious about that statement, considering `SpongeImplHooks#isMainThread` is this check, with a `isServerAvailable` check prepended to it. I had a look at SF and SV to see if we overwrite that check, and we don't.

What is the optimization check? Checking two booleans is never going to be faster than checking one, is it?

Reply:
It’s an overwrite (SpongeImplHooks is our “overwrite for optimizations and forge compat”) in ThreadChecks package

Reply:
They are both equal in performance. With that said, we do try (stress try here...) to not call API methods to keep the call stacks clean.

Biggest thing to consider is the isServerAvailable check. You want that check to happen here.

Reply:
> It’s an overwrite (SpongeImplHooks is our “overwrite for optimizations and forge compat”) in ThreadChecks package

Huh, Github didn't show that when I searched. Cool - learn something new every day. Neat.

> With that said, we do try (stress try here...) to not call API methods to keep the call stacks clean.

Fair point. I was going to change it anyway - but good to keep in mind.

Reply:
> Biggest thing to consider is the isServerAvailable check. You want that check to happen here.

That's what the `SpongeImplHooks` version will do, since it keeps itself updated whether the server is available, whether the server thread itself exists as a field, all in one method call.